### PR TITLE
Endpoint validation

### PR DESCRIPTION
**Pull request by Amuhar**

### CODE REVIEW

File: src/http/common/entities/key-query.ts [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+const toBoolean = (value): boolean => {
+  if (value === 'true') {
+    return true;
+  }
+
+  if (value == 'false') {
+    return false;
+  }
+
+  throw new BadRequestException(['used must be a boolean value']);
```
Comment (infloop) [inline src/http/common/entities/key-query.ts:15]: The name of the property is hard-coded in the function in exception. Maybe do the parameter with property name?

**Review by infloop — APPROVED** [review]

File: src/http/sr-modules-validators/entities/query.ts [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
 
 export class Query {
-  @Transform(({ value }) => Number(value))
   @ApiProperty({
     required: false,
     description: 'Number of validators to exit. If validators number less than amount, return all validators.',
   })
+  @IsInt()
+  @Min(0)
+  @Type(() => Number)
```
[the remark this answers is not in the document]
Comment (Amuhar) [reply src/http/sr-modules-validators/entities/query.ts:12]: We need it for transformation string to number , but I will double check

File: src/http/common/entities/key-query.ts (hunk 1 above)
Comment (Amuhar) [reply src/http/common/entities/key-query.ts:15]: Agree

File: src/http/sr-modules-validators/entities/query.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amuhar) [reply src/http/sr-modules-validators/entities/query.ts:12]: i checked
Without `@Type(() => Number) ` got 
```json
{
  "statusCode": 400,
  "message": [
    "operatorIndex must not be less than 0",
    "operatorIndex must be an integer number"
  ],
  "error": "Bad Request"
}

File: src/http/sr-modules/sr-modules.controller.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ export class SRModulesController {
     name: 'module_id',
     description: 'Staking router module_id or contract address.',
   })
+  // add patter check Module Id
```
[the remark this answers is not in the document]
Comment (Amuhar) [reply src/http/sr-modules/sr-modules.controller.ts]: fixed

File: src/http/common/entities/key-query.ts (hunk 1 above)
Comment (Amuhar) [reply src/http/common/entities/key-query.ts]: fixed

**Pull request merged**